sdcm/utils/common.py

```diff
@@ -2116,12 +2118,20 @@ def get_db_tables(session, ks, with_compact_storage=True):
 
     """
     output = []
-    for table in list(session.cluster.metadata.keyspaces[ks].tables.keys()):
-        table_code = session.cluster.metadata.keyspaces[ks].tables[table].as_cql_query()
+    for row in list(session.execute(f"select table_name from system_schema.tables where keyspace_name='{keyspace_name}'")):
+        try:
+            create_table_statement = node.run_cqlsh(f"describe {keyspace_name}.{row.table_name}").stdout.upper()
+        except (UnexpectedExit, Libssh2_UnexpectedExit) as err:
+            # SCT issue https://github.com/scylladb/scylla-cluster-tests/issues/7240
+            # May happen when disrupt_add_remove_dc nemesis run in parallel to the disrupt_add_drop_column
+            LOGGER.error("Failed to describe '%s.%s' table. Maybe the table has been deleted. Error: %s",
+                         keyspace_name, row.table_name, err.result.stderr)
+            continue
+
         if with_compact_storage is None:
-            output.append(table)
-        elif ("with compact storage" in table_code.lower()) == with_compact_storage:
-            output.append(table)
+            output.append(row.table_name)
+        elif ("WITH COMPACT STORAGE" in create_table_statement) == with_compact_storage:
```

Comment:
Are you sure about the text case?
Maybe better to add ".upper()" to the "create_table_statement" var to avoid possible case mismatches in some of the cql versions?

Reply:
it added above, in the line 2123